src/main/java/blusunrize/immersiveengineering/common/gui/FluidSorterMenu.java

```diff
@@ -86,7 +90,8 @@ public void receiveMessageFromScreen(CompoundTag message)
 			int side = message.getInt("filter_side");
 			int slot = message.getInt("filter_slot");
 			FluidStack newFilter = FluidStack.parseOptional(
-					Minecraft.getInstance().level.registryAccess(),
+					((MinecraftServer)LogicalSidedProvider.WORKQUEUE.get(LogicalSide.SERVER)).overworld()
+							.registryAccess(),
```

Comment:
Surely there is a cleaner way to do this? Maybe via the player?

Reply:
@BluSunrize if you have a moment please review the changes I made to this. I'd really like to go forward with some potion compatibility for this GUI.